js/GroupedBarChart.js

```diff
@@ -198,7 +216,46 @@ var GroupedBarChart = function(param)
         {
             var that = this;
 
+            /*Color array is randomized on each refresh*/
+            that.colors = _.shuffle(that.colors);
+
+            //Lighten Darken Function by Chris Coyier
+            //Used to lighten the colors of each group from left to right
+            function lightenDarkenColor(col, amt) {
+              var usePound = false;
+              if (col[0] == "#") {
+                  col = col.slice(1);
+                  usePound = true;
+              }
+              var num = parseInt(col,16);
+              var r = (num >> 16) + amt;
+              if (r > 255) r = 255;
+              else if  (r < 0) r = 0;
+              var b = ((num >> 8) & 0x00FF) + amt;
+              if (b > 255) b = 255;
+              else if  (b < 0) b = 0;
+              var g = (num & 0x0000FF) + amt;
+              if (g > 255) g = 255;
+              else if (g < 0) g = 0;
+              /*Fixed Lighten algorithm, was truncating hexes randomly.*/
+              var result = "000000" + (g | (b << 8) | (r << 16)).toString(16);
+              result = result.substr(-6);
+              result = (usePound?"#":"") + result;
+              return result;
+            }
+
+
+
+            /*The amount that bars are darkened on Mouseover*/
+            var mouseOverDarken = -20;
+            /*Reverses the effect of the darken on mouseout*/
+            var mouseOverReverse = -1 * mouseOverDarken;
+
             var tooltip_mouseover = function(d) {
+                /*Select the bar that calls mouseover and darkens the bar*/
+                var bar = d3.select(this);
+                bar.attr("fill", lightenDarkenColor(bar.attr("fill"), mouseOverDarken));
```

Comment:
I dont think it's a good user experience when the bar is darken when hovered. It's because some colors when darkened, becomes almost the same color as the one beside it. Just my two cents, maybe we should ask @JesseLeeJohnston about this one